projects/js-packages/charts/rollup.config.mjs

```diff
@@ -1,78 +1,109 @@
+	'visx/group/index': 'src/visx/group/index.ts',
+	'visx/legend/index': 'src/visx/legend/index.ts',
+	// Include provider index files
+	'providers/chart-context/index': 'src/providers/chart-context/index.ts',
+	'providers/theme/index': 'src/providers/theme/index.ts',
+};
+
+// Helper function to create consistent output configurations
+// Reduces duplication between CJS, ESM, and TypeScript declaration outputs
+const createOutputConfig = ( dir, format ) => ( {
+	dir,
+	format,
+	preserveModules: true, // Keep individual module files instead of bundling
+	preserveModulesRoot: 'src', // Remove 'src' from output paths
+	exports: 'named', // Ensure named exports are preserved for tree-shaking
+} );
+
+// Main configuration for building JavaScript modules (CJS and ESM)
 const mainConfig = {
-	input: 'src/index.ts',
+	input: inputConfig,
 	output: [
-		{
-			dir: './dist/cjs/',
-			format: 'cjs',
-			preserveModules: true,
-			preserveModulesRoot: 'src',
-			sourcemap: true,
-			sourcemapPathTransform: relativeSourcePath => `/@automattic/charts/${ relativeSourcePath }`,
-		},
-		{
-			dir: './dist/mjs/',
-			format: 'esm',
-			preserveModules: true,
-			preserveModulesRoot: 'src',
-			sourcemap: true,
-		},
+		createOutputConfig( './dist/cjs/', 'cjs' ), // CommonJS for Node.js compatibility
+		createOutputConfig( './dist/mjs/', 'esm' ), // ES modules for modern bundlers
 	],
-	external: [ 'react', 'react-dom', /^@visx\/.*/, '@react-spring/web', 'clsx', 'tslib' ],
+	// Dependencies are externalized automatically by peerDepsExternal plugin and resolveOnly option
 	plugins: [
-		peerDepsExternal( { includeDependencies: true } ),
+		peerDepsExternal( { includeDependencies: true } ), // Automatically externalize all dependencies from package.json
 		resolve( {
 			preferBuiltins: true,
-			extensions: [ '.tsx', '.ts', '.js', '.jsx' ],
+			extensions: [ '.tsx', '.ts', '.js', '.jsx' ], // Resolve these file extensions
+			// Only resolve relative imports and src/ files - this prevents bundling node_modules
+			resolveOnly: [ /^\.\.?\//, /^src\// ],
 		} ),
-		commonjs(),
-		json(),
+		commonjs(), // Convert CommonJS modules to ES modules
+		json(), // Import JSON files as modules
 		postcss( {
-			extract: 'style.css', // Generate individual CSS files
+			extract: 'style.css', // Extract CSS to a separate file
 			autoModules: true, // Automatically handle .module.scss as CSS modules
 			modules: true, // Enable CSS modules
 			use: [ 'sass' ], // Enable SCSS support
 			minimize: true, // Minify the CSS
 		} ),
 		typescript( {
 			tsconfig: './tsconfig.json',
-			declaration: false,
-			sourceMap: true,
+			declaration: false, // Don't generate .d.ts files here (handled by dtsConfig)
 			compilerOptions: {
-				verbatimModuleSyntax: true,
+				verbatimModuleSyntax: true, // Preserve import/export syntax exactly
 			},
 			exclude: [ 'node_modules', 'dist', '**/stories/**', '**/*.test.{ts,tsx}' ],
 		} ),
-		terser(),
```

Comment:
Stop minifying files, so that we don't have to include source map and the code will be minified by consumer's bundler.